cpp/common/src/codingstandards/cpp/EncapsulatingFunctions.qll

```diff
@@ -18,6 +18,39 @@ class MainFunction extends MainLikeFunction {
   }
 }
 
+/**
+ * A test function from the GoogleTest infrastructure.
+ *
+ * Such functions can be treated as valid EntryPoint functions during analysis
+ * of "called" or "unused" functions. It is not straightforward to identify
+ * such functions, however, they have certain features that can be used for
+ * identification. This can be refined based on experiments/real-world use.
+ */
+class GTestFunction extends MainLikeFunction {
+  GTestFunction() {
+    // A GoogleTest function is named "TestBody" and
+    this.hasName("TestBody") and
+    // is enclosed by a class that inherits from a base class
+    this.getEnclosingAccessHolder() instanceof Class and
+    exists(Class base |
+      base = this.getEnclosingAccessHolder().(Class).getABaseClass() and
+      (
+        // called "Test" or
+        exists(Class c | base.getABaseClass() = c and c.hasName("Test"))
+        or
+        // defined under a namespace called "testing" or
+        exists(Namespace n | n = base.getNamespace() | n.hasName("testing"))
```

Comment:
I think we want to `and` these two conditions - i.e. we are looking for a base class which is called `Test` and is in the `testing` namespace. Just looking for base classes called `Test` could result in false negatives for code which uses other frameworks.

Reply:
Requesting to review commit: [a48c7cc](https://github.com/github/codeql-coding-standards/pull/725/commits/a48c7cc6af11ed445cee5a43d1e823db6b2c3833)